libopenage/pathfinding/cost_field.h

```diff
@@ -95,15 +103,33 @@ class CostField {
 	 * Set the cost field values.
 	 *
 	 * @param cells Cost field values.
+	 * @param changed Time at which the cost value is changed.
+	 */
+	void set_costs(std::vector<cost_t> &&cells, const time::time_t &changed);
+
+	/**
+	 * Check if the cost field is dirty at the specified time.
+	 *
+	 * @param time Cost field is dirty if the cost field is accessed after the time given in valid_until.
 	 */
-	void set_costs(std::vector<cost_t> &&cells);
+	bool is_dirty(const time::time_t &time);
+
+	/**
+	 * Cleans the dirty flag by setting it to time_MAX.
```

Comment:
The internal mechanism of tracking the dirty status is not really relevant for the user of this method. Therefore, you can remove it from the docstring.

```suggestion
	 * Clear the dirty flag.
```